app/assets/javascripts/host_edit.js

```diff
@@ -1,5 +1,12 @@
 //= require parameter_override
 
+var HOST_EDIT_ATTRIBUTES = {
+  "architecture" : ['architecture_id', 'organization_id', 'location_id'],
```

Comment:
another comment, we could move this whole file to be managed under webpack, then you would have a tfm.hosts. (or similar) name space instead of an ugly global var.(and would be a lot easier to test).
I'm more than happy to help if you need guidance on how to do that.